artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -282,18 +285,52 @@ public void validateMatching(Queue queue, AMQPBrokerConnectionElement connection
    public void createLink(Queue queue, AMQPBrokerConnectionElement connectionElement) {
       if (connectionElement.getType() == AMQPBrokerConnectionAddressType.PEER) {
          Symbol[] dispatchCapability = new Symbol[]{AMQPMirrorControllerSource.QPID_DISPATCH_WAYPOINT_CAPABILITY};
-         connectSender(queue, queue.getAddress().toString(), null, null, null, null, dispatchCapability, null);
-         connectReceiver(protonRemotingConnection, session, sessionContext, queue, dispatchCapability);
-      } else {
-         if (connectionElement.getType() == AMQPBrokerConnectionAddressType.SENDER) {
-            connectSender(queue, queue.getAddress().toString(), null, null, null, null, null, null);
+         connectSender(queue, queue.getAddress().toString(), null, null, null, null, null, null, dispatchCapability);
+         connectReceiver(protonRemotingConnection, session, sessionContext, queue, null, dispatchCapability);
+      } else if (connectionElement.getType() == AMQPBrokerConnectionAddressType.SENDER) {
+         String targetAddress = null;
+         Symbol[] targetCapabilities = null;
+
+         if (connectionElement instanceof AMQPSenderBrokerConnectionElement) {
+            final AMQPSenderBrokerConnectionElement sender = (AMQPSenderBrokerConnectionElement) connectionElement;
+
+            targetAddress = sender.getTargetAddress();
+            targetCapabilities = parseTerminusCapabilities(sender.getTargetCapabilities());
          }
-         if (connectionElement.getType() == AMQPBrokerConnectionAddressType.RECEIVER) {
-            connectReceiver(protonRemotingConnection, session, sessionContext, queue);
+
+         connectSender(queue, queue.getAddress().toString(), null, null, null, null, null, targetAddress, targetCapabilities);
```

Comment:
Could we actually just move the other 'is an explicit override address defined' bit here, and then simply pass 'targetName' as before (perhaps renamed to targetAddress though?), but which is then either the queues address as originally, or the new configured 'override' target if defined? Would save passing around 2 competing values for 1 thing and leave the other code simpler / as-is.

Similarly for the receiver bit below.